File: models/objects/accounts_plans.js
var mongoose = require('mongoose'),
	Schema = mongoose.Schema,
	_ = require('underscore'),
	returnJSON = require('./utils').returnJSON;

var accounts_plans_schema = new Schema({
	id: String,
 	name: String,
	price: String,
	account_id: {type: Schema.ObjectId, private: true},
	parrots: Number,
	active: {type: Boolean, default: true, private: true},
	created_at: {type: Date, default: Date.now, private: true},
});

accounts_plans_schema.method('disable', function(callback) {
	this.active = false;
	this.save(function(err){
		if (err) calback(err);
		else callback(null);
	});
});

mongoose.model('AccountsPlans', accounts_plans_schema);
var AccountsPlans = mongoose.model('AccountsPlans');
module.exports = AccountsPlans;

AccountsPlans.create_from_plan = function(account, new_plan, callback) {
	var plan_data = new_plan.toJSON();
	delete plan_data._id;
	plan_data['active'] = true;
	plan_data['account_id'] = account._id;
	var new_account_plan = new AccountsPlans(plan_data);
	new_account_plan.save(function(err){
		if(err) callback(err);
		else {
			callback(null, new_account_plan);
		}
	});
};


AccountsPlans.prototype.returnJSON = returnJSON;
AccountsPlans.prototype.returnJSON = returnJSON;

File: dal/payparrot_dal/accounts_plans.py
from datetime import datetime
from hashlib import sha256
from random import random
from time import time

from payparrot_dal.base import BaseModel

class AccountsPlans(BaseModel):
    _meta = {
        'collection': 'accounts_plans',
        'fields': {
            'name': {'required': True},
            'price': {},
            'parrots': {},
            'account_id': {},
            'active': {'default': True, 'private': True},
            'created_at': {'private': True, 'default':datetime.now}
        }
    }

    def disable(self):
        self.update({'active': False})

    @classmethod 
    def create_from_plan(self,db,account,new_plan):
        if new_plan['id']:
            del new_plan['id']
        new_plan['active'] = True
        new_plan['account_id'] = account.get("id")
        new_account_plan = AccountsPlans(db, new_plan)
        new_account_plan.insert()
        return new_account_plan
